### ISSUE TITLE
new source for shaders (and addons if support for managing them is added)

### ISSUE THREAD

**Issue by zany130** [sonic2kk/steamtinkerlaunch#1182]
## System Information
- SteamTinkerLaunch version:  `steamtinkerlaunch-v14.0.20240904-1`
- Distribution: Bazzite 40

## Feature Description
there is a more up-to-date list of reshade shader repositories at https://github.com/crosire/reshade-shaders/tree/list

There is even a list of reshade add-ons there (if we ever add support for that. I am considering matching the official installer functionality, which allows you to download shaders/addons, but the actual reshade_addon version you need to download yourself (as that will trip anticheat)). I *think* this is what the Reshade Windows installer uses

**Comment by sonic2kk** [comment]
I'm not sure how any of our current Shader Repo stuff is implemented but I'm open to an improvement here. I don't use (and quite dislike) ReShade so I don't think I will implement this, but I'm open to it being implemented so long as whoever implements it confirms that it works on their end (I am fine with changes in behaviour if they are an improvement, so long as someone implementing it is familiar with a desired flow :smile:)

Since that looks like a regular markdown wiki page we could probably just `wget` and parse that to get the shader links, and store them in whatever format we currently do.

I'm not sure of the specifics of the current implementation, the most I have ever done is make sure the DLLs go into the right folder with the right name and I have not touched the shader repo stuff for a *very* long time (2+ years at this point and it was probably by accident the last time). But so long as an implementation is submitted with good testing instructions and that doesn't break whatever is currently in place, I have no objections to this being added. It just won't come from me ;-)

**Comment by zany130** [comment]
yeah no worries. I just wanted to make an issue for it so I don't forget. I hoping that is would be as straight foward as changing the URL for the reshade repo list. I think the URL can already be changed by the user in the URL page of steamtinkerlaunch so ill just start with testing there .(forgot what it was called but I remember there being a page where you can change the URL of several dependices of STL)

The addon part would be a little more complex and would probably be another PR I do latter down the road (prob not anytime soon)

I not even really sure we should have a downloader for addons but since the official installer does I was thinking we should match it.



_____________________________________________________________________________________
for some context since you probably don't use reshade much 😆 

Addons are extra .dll that end in `.addon` that you can add the game directory and they add extra functionality to reshade. And they need the addon version of reshade which allows reshade to work on networked games (by default reshade disables itself if it detects a network connection in the game)

the main one being https://github.com/4lex4nder/ReshadeEffectShaderToggler for controlling the rendering order of shaders
